src/server/worker/worker.go

```diff
@@ -73,21 +74,25 @@ func NewWorker(
 	worker.APIServer = server.NewAPIServer(driver, worker.status, env.Config().PodName)
 
 	go worker.master(env)
-	go worker.worker()
+	go worker.worker(env)
 	return worker, nil
 }
 
-func (w *Worker) worker() {
+func (w *Worker) worker(env serviceenv.ServiceEnv) {
 	ctx := w.driver.PachClient().Ctx()
 	logger := logs.NewStatlessLogger(w.driver.PipelineInfo())
 
 	backoff.RetryUntilCancel(ctx, func() error {
 		eg, ctx := errgroup.WithContext(ctx)
 		driver := w.driver.WithContext(ctx)
+		imageID, err := getUserContainerImageID(ctx, env)
```

Comment:
since the Image ID is static for the lifetime of the Worker process, I think it might make sense to load and reference it through the driver

Reply:
This makes sense. I've moved the function into the driver and made it more generic for container image IDs